bean-validation

girirajsharma opened the pull request:
Added comments(description) for bean validation annotations and added two new tests for bean validation in MemberValidationTest.java. Please ignore commit 5dee8a3.

girirajsharma committed:
Added comments(description) for all built-in constraints used for Member
bean validation.Also,added dateOfBirth,eventDate,minQuantity and
maxQuantity as additional fields to implement several other constraints
using @Past,@Future,@Min,@Max as annotations.

Signed-off-by: girirajSharma <giriraj.sharma27@gmail.com>

girirajsharma committed:
Added comments(description) for bean validation annotations.Also, added
dateOfBirth,eventDate,minQuantity,maxQuantity as new fields with
@Past,@Future,@Min,@Max as respective annotations.

Signed-off-by: girirajSharma <giriraj.sharma27@gmail.com>

girirajsharma committed:
Added two new tests for bean validation namely testDateOfBirthViolation
and testMinQuantityViolation to test for constraints on dateOfBirth and
minQuantity fields of Member.Also,createValidMember method was modified
to initialize new fields of Member class object(member) with valid
values.

Signed-off-by: girirajSharma <giriraj.sharma27@gmail.com>

girirajsharma committed:
Changed ${version.jboss.bom.eap} from 6.2.0-build-7 to 6.2.1.GA as it is
the latest version available from the
jboss-ga-repository(http://maven.repository.redhat.com/techpreview/all).

girirajsharma committed:
Added comment(description) for @Column annotation used to specify a
mapped column for a persistent property or field.

girirajsharma committed:
Added date_of_birth, event_date, min_quantity and max_quantity as new
column names into the Table(MEMBER_BEAN_VALIDATION).

girirajsharma committed:
Added date_of_birth, event_date, min_quantity and max_quantity as new
column names into the Table(MEMBER_BEAN_VALIDATION).

girirajsharma committed:
Replaced 4 violations with 8 as a Member instance initialized with
default constructor will fail to validate against 8 Not Null fields.

girirajsharma committed:
Added imports (javax.validation.constraints.*) for validation
constraints Past,Future,Max and Min.

girirajsharma committed:
bean-validation-custom-constraints : Bean validation using custom
constraints via Arquillan Example
This quickstart will show you how Bean Validation API can be used to
create custom constraints for validation of data and how to use
arquillan to test the same.

Signed-off-by: girirajSharma <giriraj.sharma27@gmail.com>

sgilda commented on bean-validation-custom-constraint/README.md:
```diff
@@ -0,0 +1,102 @@
+bean-validation-custom-constraints: Bean Validation using custom constraints via Arquillian Example
```
bean-validation-custom-constraint: Bean Validation using custom constraints via Arquillian Example

(The quickstart name must match the folder name)

sgilda commented on bean-validation-custom-constraint/README.md:
```diff
@@ -0,0 +1,102 @@
+Author: Giriraj Sharma  
+Level: Beginner  
+Technologies: Bean Validation, JPA  
+Summary: Shows how to use Arquillian to test Bean Validation using custom constraints 
```
You need 2 spaces at the end of this line so it renders a carriage return.

sgilda commented on bean-validation-custom-constraint/pom.xml:
```diff
@@ -0,0 +1,250 @@
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>jboss-eap-bean-validation-customConstraint</groupId>
```
This should be:

```
    <groupId>org.jboss.quickstarts.eap</groupId>
```

sgilda commented on bean-validation-custom-constraint/pom.xml:
```diff
@@ -0,0 +1,250 @@
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>jboss-eap-bean-validation-customConstraint</groupId>
+  <artifactId>jboss-eap-bean-validation-customConstraint</artifactId>
```
This should be:

```
    <artifactId>jboss-bean-validation-custom-constraint</artifactId>
```

sgilda commented on bean-validation-custom-constraint/pom.xml:
```diff
@@ -0,0 +1,250 @@
+
+  <groupId>jboss-eap-bean-validation-customConstraint</groupId>
+  <artifactId>jboss-eap-bean-validation-customConstraint</artifactId>
+  <version>0.0.1-SNAPSHOT</version>
```
```
    <version>6.2.0-redhat-SNAPSHOT</version>
```

sgilda commented on bean-validation-custom-constraint/pom.xml:
```diff
@@ -0,0 +1,250 @@
+  <groupId>jboss-eap-bean-validation-customConstraint</groupId>
+  <artifactId>jboss-eap-bean-validation-customConstraint</artifactId>
+  <version>0.0.1-SNAPSHOT</version>
+  <packaging>jar</packaging>
```
Doesn't this generate a WAR?

sgilda commented on bean-validation-custom-constraint/pom.xml:
```diff
@@ -0,0 +1,250 @@
+  <version>0.0.1-SNAPSHOT</version>
+  <packaging>jar</packaging>
+
+  <name>JBoss EAP Quickstart: bean-validation-customConstraint</name>
```
```
    <name>JBoss EAP Quickstart: bean-validation-custom-constraint</name>
```

sgilda commented on bean-validation-custom-constraint/pom.xml:
```diff
@@ -0,0 +1,250 @@
+  <packaging>jar</packaging>
+
+  <name>JBoss EAP Quickstart: bean-validation-customConstraint</name>
+    <description>An Arquillian Bean Validation by Custom Constraints test example</description>
```
```
    <description>An example that demonstrates Arquillian Bean Validation Custom Constraints </description>
```

sgilda commented on bean-validation-custom-constraint/pom.xml:
```diff
@@ -0,0 +1,250 @@
+        <version.jboss.maven.plugin>7.4.Final</version.jboss.maven.plugin>
+
+        <!-- Define the version of the JBoss BOMs we want to import to specify tested stacks. -->
+        <version.jboss.bom.eap>6.2.1.GA</version.jboss.bom.eap>
```
```
    <version.jboss.bom.eap>6.2.0-build-7</version.jboss.bom.eap>
```

girirajsharma replied:
@sgilda @LightGuard When I reset jboss.bom.eap version to 6.2.0-build-7, I face several build errors like  "Project build error: Non-resolvable import POM: Failure to find org.jboss.bom.eap:jboss-javaee-6.0-with-tools:pom:6.2.0-build-7 in http://maven.repository.redhat.com/earlyaccess/all/ was cached in the local repository, resolution will not be reattempted until the update interval of jboss-earlyaccess-repository has elapsed or updates are forced "  in dependency management.

sgilda commented on bean-validation-custom-constraint/README.md:
```diff
@@ -0,0 +1,102 @@
+
+This project demonstrates how to use CDI 1.0, JPA 2.0 and Bean Validation 1.0. Bean Validation API allows the developers to define their own constraints by creating a new annotation and writing the validator which is used to validate the value. This quickstart will show you how to create custom constraints and 
+
+then use it to validate your data. It includes a persistence unit and some sample persistence code to introduce you to database access in enterprise Java. This quickstart does not contain a user interface layer. 
```
I'm not sure why there is a blank line breaking up this sentence.

sgilda replied:
This shouldn't break between 'custom constraints and ' ... 'then use it to validate'

sgilda commented:
Nice job on this!  

There are a few minor issues I discovered running the QS Tools. I noted them in the README and POM files. See step 8 under 'Basic Steps' in the Contributing Guide for details on how to run the QS Tools report. The problems with the POM file are also causing the Arquillian tests to fail.

After you fix the issues I mentioned, would you mind squashing the commits down to 1 commit? That makes it easier to review.

When you fix the issues, I will get someone to do a code review.

Thanks for submitting this quickstart!

sgilda commented:
Even when I fix up your POM file, I am still getting an error when I run the tests. I see this in the server log:

13:26:26,152 ERROR [org.jboss.as.server](management-handler-thread - 5) JBAS015870: Deploy of deployment "test.war" was rolled back with the following failure message: 
{"JBAS014671: Failed services" => {"jboss.persistenceunit.\"test.war#primary\"" => "org.jboss.msc.service.StartException in service jboss.persistenceunit.\"test.war#primary\": javax.persistence.PersistenceException: [PersistenceUnit: primary] Unable to build EntityManagerFactory
    Caused by: javax.persistence.PersistenceException: [PersistenceUnit: primary] Unable to build EntityManagerFactory
    Caused by: org.hibernate.AnnotationException: No identifier specified for entity: org.jboss.as.quickstarts.bean_validation_customConstraint.MyPerson"}}

girirajsharma commented:
@sgilda -Thanks a lot for reviewing and pointing out my errors.I shall try to fix all the bugs , you have mentioned. I will then perform testing using QS tools and will also see to it which problems in POM file are causing arquillian tests to fail. Once I fix the issues you mentioned, I would squash all the commits down to 1 commit so as to make it easy for code reviewer. Between,I also tried to extend the functionality of bean-validation quickstart(one which is already present in git master branch) in my commits (5dee8a3, a932f63, 341d79c, b1f898a, 9d93269, 7df11bb). Please review those commits also by a code reviewer whenever you feel like as per your convenience and time.Thanks.

LightGuard commented on bean-validation-custom-constraint/src/main/java/org/jboss/as/quickstarts/bean_validation_custom_constraint/AddressValidator.java:
```diff
@@ -0,0 +1,54 @@
+
+        if (value.getCity() == null || value.getCountry() == null || value.getLocality() == null
+            || value.getPinCode() == null || value.getState() == null || value.getStreetAddress() == null) {
+            return false;
```
Shouldn't you test for empty too?

girirajsharma replied:
@LightGuard Thanks for pointing out. I shall make necessary changes and will review the code once again followed by testing.

girirajsharma committed:
Declared the proxy server settings in Maven configuration file
contributor-settings.xml to enable downloading of dependencies through
Maven in case of proxy servers.

girirajsharma committed:
bean-validation-custom-constraints : Bean validation using custom
constraints via Arquillan Example.
This quickstart will show how Bean Validation API can be used to
create custom constraints for validation of data and how to use
arquillan to test the same.

Signed-off-by: girirajSharma <giriraj.sharma27@gmail.com>

girirajsharma committed:
Added License Information and Contributor Agreement to pom.xml

Signed-off-by: girirajSharma <giriraj.sharma27@gmail.com>

girirajsharma committed:
bean-validation-custom-constraints
Bean validation using custom constraints via Arquillan Example
This quickstart will show you how Bean Validation API can be used to
create custom constraints for validation of data and how to use
arquillan to test the same.Please ignore all previous commits in
#858.Current commit is wrapped up with entire quickstart source code.

Signed-off-by: girirajSharma <giriraj.sharma27@gmail.com>

sgilda commented:
@girirajsharma : I just updated to get your latest changes and I get errors when I run `mvn clean test -Parq-jbossas-remote`

Here are the results:

Running org.jboss.as.quickstarts.bean_validation_custom_constraint.MyPersonTest
Tests run: 5, Failures: 4, Errors: 0, Skipped: 0, Time elapsed: 3.231 sec <<< FAILURE!

Results :

Failed tests:   testAddressViolation(org.jboss.as.quickstarts.bean_validation_custom_constraint.MyPersonTest): Address Field  was invalid expected:<Address Fields must not be null/empty and obey character limit constraints> but was:<null>
  testRegisterEmptyPerson(org.jboss.as.quickstarts.bean_validation_custom_constraint.MyPersonTest): Three violations were found expected:<Name and Address fields must not be null/empty> but was:<null>
  testFirstNameSizeViolation(org.jboss.as.quickstarts.bean_validation_custom_constraint.MyPersonTest): First Name was invalid expected:<size must be between 4 and 2147483647> but was:<3>
  testFirstNameNullViolation(org.jboss.as.quickstarts.bean_validation_custom_constraint.MyPersonTest): First Name was invalid expected:<may not be null> but was:<null>

Tests run: 5, Failures: 4, Errors: 0, Skipped: 0

sgilda commented:
I also noticed a few minor things in the README, but I can take care of them when everything else is complete.

Can you also squash the commits after you've made the fixes? Thanks!

girirajsharma commented:
@sgilda The above logs actually indicate test failures in cases where either Person or Address entity beans are not set as per custom constraints. I will modify MyPersonTest.java by setting entity beans in such a way so that it doesn't gives any test failures and results in successful build.

girirajsharma committed:
Set valid values for Person and Address entity beans as per custom
constraints so that mvn clean install results in a successful build.

Signed-off-by: girirajSharma <giriraj.sharma27@gmail.com>

sgilda commented:
Closing this pull. @girirajsharma will issue a new pull.
